keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+     *
+     * @return A Flux of Secrets. The Flux contains all of the secrets in the vault.
+     */
+    public Flux<SecretInfo> listSecretsAsync() {
+        Mono<RestResponse<Page<SecretInfo>>> result = service.getSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE);
+
+        return getPagedSecrets(result);
+    }
+
+    /**
+     * Lists {@link DeletedSecret} secrets of the key vault.
+     * The Get Deleted Secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @return A Flux of Deleted Secrets. The Flux contains all of the deleted secrets in the vault.
+     */
+    public Flux<DeletedSecret> listDeletedSecretsAsync() {
+        Mono<RestResponse<Page<DeletedSecret>>> result = service.getDeletedSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE);
+        return getPagedDeletedSecrets(result);
+    }
+
+    /**
+     * List all versions of the specified secret. The individual secret version response in the list is represented by {@link SecretInfo}
+     * The full secret identifier and attributes are provided in the response. No values are returned for the secrets.
+     * This operations requires the secrets/list permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException thrown if name parameter is null
+     * @return A Flux of Secrets. The Flux contains all of the versions of the secret in the vault.
+     */
+    public Flux<SecretInfo> listSecretVersionsAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        Mono<RestResponse<Page<SecretInfo>>> result = service.getSecretVersions(vaultEndPoint, name, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE);
+
+        return getPagedSecrets(result);
+    }
+
+    /**
+     * List secrets in a specified key vault.
+     * The Get Secrets operation is applicable to the entire vault.
+     * However, only the base secret identifier and its attributes are provided in the response. Individual secret versions are not listed in the response.
+     * This operation requires the secrets/list permission.
+     *
+     * @param maxPageResults Maximum number of results to return in a page. If not specified, the service will return up to 25 results.
+     * @throws NullPointerException thrown if maxPageResults parameter is null
+     * @return A Flux of Secrets. The Flux contains all of the secrets in the vault.
+     */
+    public Flux<SecretInfo> listSecretsAsync(int maxPageResults) {
+        Objects.requireNonNull(maxPageResults, "The maximum page results parameter cannot be null.");
+        Mono<RestResponse<Page<SecretInfo>>> result = service.getSecrets(vaultEndPoint, maxPageResults, API_VERSION, ACCEPT_LANGUAGE);
+
+        return getPagedSecrets(result);
+    }
```

Comment:
I don't believe we want these `maxPageResults` parameters in public API...is that correct @schaabs? Instead we should just internally page as appropriate for the service.

Reply:
Default Page size is 25.
if user specifies custom page size, it controls the number of pages with nextPageLink get concatenated together in Flux.
The only benefit would be if user provides larger than 25 page size, then that would reduce the rest calls made to AZ KV.
But in all cases all the secrets will be listed and processed.

Reply:
I want @schaabs to weigh in on this.

Reply:
We've concluded this discussion now - they should be removed.